historyki_roblox/resource_manager.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+import random
+
+from PIL import ImageFont, Image
+
+
+class ResourceManager:
+
+    def __init__(self):
+        self.root_path = os.path.dirname(os.path.dirname(__file__))
+
+    # CHARACTER
+
+    def get_list_of_characters(self):
+        return os.listdir(f'{self.root_path}/data/characters')
+
+    def get_roblox_character(self, requested_gender: str) -> str:
+        chosen_gender = 'f' if requested_gender == 'FEMALE' else 'm'
+        chosen = random.choice([char for char in self.get_list_of_characters() if chosen_gender in char])
+        return f'{self.root_path}/data/characters/{chosen}'
+
+    # OSKAREK
+
+    def get_list_of_oskareks(self):
+        return os.listdir(f'{self.root_path}/output/oskareks')
+
+    def save_oskarek_image(self, image: Image, image_name: str):
+        return image.save(f'{self.root_path}/output/oskareks/{image_name}')
+
+    def get_oskarek_image(self, requested_gender: str) -> str:
+        chosen_gender = 'f' if requested_gender == 'FEMALE' else 'm'
+        chosen = random.choice([char for char in self.get_list_of_oskareks() if chosen_gender == char[0]])
+        return f'{self.root_path}/output/oskareks/{chosen}'
+
+    # THUMBNAIL
+
+    def get_list_of_backgrounds(self):
+        return os.listdir(f'{self.root_path}/data/thumbnail/background')
+
+    def get_list_of_emoji(self):
+        return os.listdir(f'{self.root_path}/data/thumbnail/emoji')
+
+    def get_thumbnail_data(self):
+        with open(f"{self.root_path}/data/thumbnail/thumbnail_data.json") as fp:
+            data = json.load(fp)
+        return data
+
+    def get_thumbnail_background(self, resize_=None, filter_=None):
+        background_image = Image.open(random.choice(self.get_list_of_backgrounds()))
+        if resize_:
+            background_image = background_image.resize(resize_)
+        if filter_:
+            background_image = background_image.filter(filter_)
+        return background_image
+
+    # possible categories: WOW, LAUGH, SAD etc...
+    def get_thumbnail_emoji(self, category: str, resize_=None):
```

Comment:
If you want to have a predefined list of categories its good idea to use enum class [https://docs.python.org/3/library/enum.html]
```
class EmojiCategory(Enum):
    WOW = "w"
    
... 

    def get_thumbnail_emoji(self, category: EmojiCategory, resize_=None):
        ...
        chosen_category = category.value
```
Then in the thumbnail_builder we could import the EmojiCategory and use it while calling for an emoji like this:
```
get_thumbnail_emoji(EmojiCategory.WOW)
```
without a need to check whether the passed category name is a valid string or not

Reply:
added additional Enum class according to your comment